code/modules/admin/admin_verbs.dm

```diff
@@ -902,12 +902,27 @@ var/list/admin_verbs_ticket = list(
 	if(!check_rights(R_ADMIN))
 		return
 
-	prefs.toggles ^= CHAT_ATTACKLOGS
-	prefs.save_preferences(src)
-	if(prefs.toggles & CHAT_ATTACKLOGS)
-		to_chat(usr, "You now will get attack log messages")
+	if(prefs.atklog == ATKLOG_ALL)

```

Comment:
To implement a category system, you can use an `input` prompt here with the list of all logging categories to allow individual toggling, plus a few additional presets `ALL`, `CRITICAL`, `PLAYERS`, etc. that overwrite the `atklog` pref on the admins datum, which could now be implemented as a list.